_docs/ux-design.md

```diff
@@ -0,0 +1,344 @@
+instrumentation tool allows co-operation between each of these entities without
+coupling them too tightly.
+
+## User Experience Overview
+
+The OpenTelemetry Go compile-time instrumentation tool is composed of the
+following software artifacts (in the form of Go packages):
+
+- `github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel`,
+  a command-line tool that can be installed using `go install` or using
+  `go get -tool`;
+- `github.com/open-telemetry/opentelemetry-go-compile-instrumentation/runtime`,
+  a small, self-contained package that contains essential runtime functionality
+  used by instrumented applications (not intended for manual usage).
+
+### Getting Started
+
+The tool offers a wizard for getting started with the tool automatically by
+following a series of prompts:
+
+```console
+$ go run github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel@latest setup
+╭──────────────────────────────────────────────────────────────────────╮
+│                                                                      │
+│           OpenTelemetry compile-time instrumentation tool            │
+│                    v1.2.3 (go1.24.1 darwin/arm64)                    │
+│                                                                      │
+╰──────────────────────────────────────────────────────────────────────╯
+✨ This setup assistant will guide you through the steps needed to properly
+   configure gotel for the github.com/example/demo application. After you've
+   answered all the prompts, it will present you with the list of actions it
+   will execute; and you will have an choice to apply those or not.
+🤖 Press enter to continue...
+
+ℹ️ Registering gotel as a tool dependency is recommended: it allows you to
+   manage the dependency on gotel like any other dependency of your application,
+   via the go.mod file. When using go 1.24 or newer, you can use `go tool gotel`
+   to use the correct version of the tool.
+   Not registering a go tool dependency allows instrumenting applications
+   without modifying their codebase at all (not even the `go.mod` file); which
+   may be preferred for building third-party applications or integrating in the
+   CI/CD pipeline. The reproductibility of builds is no longer guaranteed by the
+   go toolchain, and the application may be built with newer versions of
+   dependencies than those in the `go.mod` file if the enabled instrumentation
+   package(s) requires it.
+🤖 Should I add gotel as a tool dependency?
+   [Yes]  No
+🆗 I will add a tool dependency
+
+ℹ️ You may enable one or more instrumentation packages for your application.
+   Most users need to choose only one. This can be changed at any time.
+🤖 What instrumentation do you want to enable for this project? (Select one or
+   more using space, then press enter to confirm your selection)
+   [X] OpenTelemetry (github.com/open-telemetry/opentelemetry-go)
```

Comment:
I think we should only provide OpenTelemetry instrumentation here. Vendor-related content probably shouldn't be here

Reply:
+1. We may provide a way for downstream projects to customize this process.